client/notebooks/Demo_Client_Notebook_Smartnoise-SQL.ipynb

```diff
@@ -165,633 +164,456 @@
-    "\n",
-    "It is recommended to use very high values on the budget parameters here to have little noise and small difference between the exact local result and the 'little noisy' server result. \n",
-    "\n",
-    "Also, make sure to use the same values of number of rows and seed to have the same dummy datasets."
+    "### Estimate cost of a query\n",
+    "Dr. Antartica checks the budget that computing the average bill length will really cost her if she asks the query with an `epsilon` and a `delta`."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 9,
-   "id": "3946425d",
+   "execution_count": 12,
+   "id": "133020c6",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Average bill length in mm\n",
-    "QUERY = \"SELECT AVG(bill_length_mm) AS avg_bill_length_mm FROM df\""
+    "cost = client.estimate_smartnoise_sql_cost(\n",
+    "    query = QUERY, \n",
+    "    epsilon = EPSILON, \n",
+    "    delta = DELTA,\n",
+    ")"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 10,
-   "id": "90cf2a6d",
+   "execution_count": 13,
+   "id": "ff19802d-cb39-48ee-9874-340a4bf2cc31",
    "metadata": {},
    "outputs": [
     {
      "data": {
       "text/plain": [
-       "47.51229381350249"
+       "'This query would actually cost her 1.0 epsilon and 5.000000000032756e-06 delta.'"
       ]
      },
-     "execution_count": 10,
+     "execution_count": 13,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "# On the remote server dummy dataframe\n",
-    "res = client.smartnoise_sql_query(\n",
-    "    query = QUERY,  \n",
-    "    epsilon = 100.0, # make sure to select high values of epsilon and delta to have small differences\n",
-    "    delta = 2.0,    # make sure to select high values of epsilon and delta to have small differences\n",
-    "    dummy = True, \n",
-    "    nb_rows = NB_ROWS,\n",
-    "    seed = SEED\n",
-    ")\n",
-    "res_server_dummy = res['query_response'][\"avg_bill_length_mm\"][0]\n",
-    "res_server_dummy"
+    "f'This query would actually cost her {cost[\"epsilon_cost\"]} epsilon and {cost[\"delta_cost\"]} delta.'"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "bb3fa8eb",
+   "id": "c255d210-7ba1-4152-8a30-97c7289dd361",
    "metadata": {},
    "source": [
-    "She then checks that the responses on the dummy locally and the dummy on the server are close enough (difference would be only due to small noise addition)."
+    "This is actually the double than what she put in input. In the background, Smartnoise-SQL decomposes the DP query in multiple other queries and the budget given in input is spent on each of these sub-queries. Here for the average, we need a sum divided by a count, hence `EPSILON` is spent once for the sum and then once more for the count. (see NOTE below for tips and explanation)."
```

Comment:
- This is actually twice as much as what she initially put in.
- And the budget given as input.